Extension/test/unit/eventing.test.ts

```diff
@@ -0,0 +1,255 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { ok, strictEqual } from 'assert';
+import { beforeEach, describe, it } from 'mocha';
+import { Async } from '../../src/Utility/Async/constructor';
+import { sleep } from '../../src/Utility/Async/sleep';
+import { Descriptors } from '../../src/Utility/Eventing/descriptor';
+import { notify, notifyNow, on, reset, subscribe } from '../../src/Utility/Eventing/dispatcher';
+import { Emitter } from '../../src/Utility/Eventing/emitter';
+import { Cancelled, Continue, EventData } from '../../src/Utility/Eventing/interfaces';
+import { is } from '../../src/Utility/System/guards';
+
+export class SomeBase extends Emitter {
+
+}
+
+class Something extends SomeBase {
+    constructor() {
+        super();
+        this.descriptors.add('server', 'mysrv');
+    }
+
+    // emitter declarations!
+    readonly initialize = this.newEvent('initialize');
+    readonly bump = this.newEvent('bump');
+
+    async init() {
+        console.debug('before initialize');
+        // tell them we are initializing
+        if (is.cancelled(await this.initialize())) {
+            // a handler has cancelled this event.  we should stop
+            console.debug('cancelled?');
+            this.wasCancelled = true;
+            return;
+        }
+        this.initWasSuccessful = true;
+        console.debug('after initialize');
+    }
+    wasCancelled: boolean = false;
+    initWasSuccessful: boolean = false;
+    initHandlerRan: boolean = false;
+
+    // event handler for the init event for this instance only
+    async 'this initialize'() {
+        this.initHandlerRan = true;
+        console.debug('init handler that is bound to the instance');
+
+        return Cancelled;
+    }
+}
+
+// make an async constructor for the Something class
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const AsyncSomething = Async(Something);
+
+describe('Event Emitters', () => {
+
+    beforeEach(() => {
+        reset();
+        // uncomment the following line to show debug messages in the console.
+        void on('debug', async (event: EventData) => { console.debug(event.text); });

```

Comment:
The comment says `uncomment`, but it's already uncommented? Is the comment wrong or should the line be commented out?

Reply:
fixed